client/image_load.go

```diff
@@ -12,12 +13,28 @@ import (
 // ImageLoad loads an image in the docker host from the client host.
 // It's up to the caller to close the io.ReadCloser in the
 // ImageLoadResponse returned by this function.
-func (cli *Client) ImageLoad(ctx context.Context, input io.Reader, quiet bool) (image.LoadResponse, error) {
+//
+// Platform is an optional parameter that specifies the platform to load from
+// the provided multi-platform image. This is only has effect if the input image
+// is a multi-platform image.
+func (cli *Client) ImageLoad(ctx context.Context, input io.Reader, opts image.LoadOptions) (image.LoadResponse, error) {
 	v := url.Values{}
 	v.Set("quiet", "0")
-	if quiet {
+	if opts.Quiet {
 		v.Set("quiet", "1")
 	}
+	if opts.Platform != nil {
+		if err := cli.NewVersionError(ctx, "1.47", "platform"); err != nil {
+			return image.LoadResponse{}, err
+		}
+
+		p, err := json.Marshal(*opts.Platform)
+		if err != nil {
+			return image.LoadResponse{}, err
```

Comment:
Should we return a `errdefs` error here? Also probably would be good to wrap the error as we did for the other one (`invalid platform:`)

Reply:
Not sure - other usages of `json`'s `Marshal/Unmarshal/NewDecoder/NewDecoder` don't return a raw error directly so I went with the same approach.

Reply:
Seems fine not to wrap it.

Reply:
Yeah, we could probably make a pass at typing those errors.

As to the wrapping; it was because I noticed that for ImageHistory we add a prefix;


```go
return nil, fmt.Errorf("invalid platform: %v", err)
```

Without that, the error may not give a clue where the problem is (just a plain JSON marshal error).